RELEASE.md

```diff
@@ -0,0 +1,61 @@
+# Release
+
+The kubernetes-base-addons repository is intended to support a minimal base set of Addons required to provide monitoring, logging, alerting, and backups for every version of Kubernetes actively supported by the upstream Kubernetes community.
+It is not intended to be tied to any specific version of Kubernetes installer (ie. Konvoy or Kommander), or Kubeaddons.
+Changes to Konvoy or Kommander to create a resource needed by an Addon should be avoided.
+
+- [Schedule](#schedule)
+- [Considerations](#considerations)
+  - [Kubernetes](#kubernetes)
+  - [Branches and Tags](#branches-and-tags)
+- [Process](#process)
+  - [Testing Release (Weekly, Thursday)](#testing-release-weekly-thursday)
+  - [Stable Release (Weekly, Wednesday)](#stable-release-weekly-wednesday)
+
+## Schedule
+
+Releases should be weekly on Wednesdays, or as needed to address CVEs.
```

Comment:
I do think weekly is too aggressive as we need to test/soak and support all those versions. I hope all those can be automated eventually, but we're not there yet. So I suggest start slow, like bi-weekly releases or even monthly.

cc @samba @chrisgaun

Reply:
I agree, I think weekly is probably too much. Remember konvoy "maintenance" releases (1.3.X) will happen as needed based on cops or others bug we want to ship fixes for.  1.4-beta releases will happen on a sprintly cadence.

Reply:
we should keep release branches in a releasable state all the time, once this goal is accomplished, we can easily cut released whenever we need/want. IMHO this is mostly a matter of good CI on PRs. one still can (and should) do final manual tests before cuttong/publishing a release

Reply:
+1 for biweekly to match the konvoy cadence

Reply:
This repo's release cadence isn't intended to match the konvoy releases. I'm ok with moving to biweekly, but I don't agree with konvoy being a reason.

The longer the release cadence, the more cognitive load is required: more changes, more discussion, more controversy, etc.

Even with the best possible suite of tests, we can’t guarantee a bug free release, but what we can do is fix those bugs quickly and get the fixes to users as soon as possible.

With a traditional slow release cycle you can still release a patch for serious issues, but less severe bugs tend to get pushed to the next release. For users experiencing those bugs, it can be very frustrating if they’re not fixed in a timely manner and can lead to an impression that the application isn’t actively maintained.

QA is another area where frequent releases actually helps. If you’re doing infrequent releases, there can be a lot of features, code and bugs added since the last release. This means that when an issue is found, it can be hard to trace its origins. On the other hand, if there have only been a relatively small amount of changes since the last release, then it makes it much easier to identify when and where a problem originated.

People also have a tendency to find bugs just before the build is due for release. With a slow release cycle this means that we generally need to delay even further until all known bugs are fixed. 

With a fast release cycle however, it’s much easier to persuade people to let the release go ahead as the bug can be fixed for the next release, which will be along soon. This allows us to use a rule of thumb is that if it’s not a regression (i.e. the bug already exists in the current release) then it shouldn’t hold up a new release. To do so would only deprive users of fixes for other issues that have been made since the last release.

Reply:
The two need not be so closely locked. We should work to maintain, and indeed enhance, the decoupling of the two. 

Konvoy needs to be able to handle upgrades of Addons without upgrading itself. Technically this is already feasible, just requires manual edits of the config file. We should simplify/automate this workflow, but that’s out of scope for this document. Main point: decoupling upgrades overall. 

Addons on a faster cadence can better facilitate discovery of issues in soak, provided soak deployment is linked to the (weekly) release process for addons. This will reduce the risk of shipping unknown bugs in Konvoy. (Naturally we need more robust testing and monitoring against soak.)

Further, as we’re leveraging these addons in other Kubernetes with Kommander, not just konvoy, the release process of Addons needs to be considered independently of Konvoy. It will effect customers on Kommander who may use mostly non-Konvoy clusters. In both Konvoy and non-Konvoy clusters, we also want to enable (optional) automatic upgrades *without upgrading konvoy*, so binding the release cadence to Konvoy is not inherently beneficial. 

Adopting a faster cycle here gets us better quality, and through automation (better than we have now) forces lower transaction cost of rapid responses, such as CVE updates.

Reply:
We should treat `Kubernetes-base-addons` as a product by itself because (as Sam B said) it could exists independent of Konvoy. Konvoy happens to be one of its consumers. However, for release cadence, we need to also consider soak time. In the past, we settled on 2 weeks soaking to discover any bug. We either need to reduce that to 1 week or move the release cadence to 2 weeks

Reply:
Seems reasonable. The 3rd option would be to have overlapping soak clusters where testinging would be deployed every week. Two weeks later, that tag would be released while last week continues to soak.